server/utils/stigLibrary.ts

```diff
@@ -0,0 +1,291 @@
+import * as fs from "fs";
+import * as path from "path";
+import AdmZip from "adm-zip";
+import { DateTime } from "luxon";
+import { Op, UniqueConstraintError } from "sequelize";
+import {
+  Assessment,
+  AssessmentItem,
+  StigData,
+  StigLibrary,
+  System,
+  Stig,
+  StigAlias,
+} from "../../db/models";
+import { hashFile } from "./hash";
+import { createBlankAssessment } from "./assessments";
+
+export type ProcessLibraryResults = {
+  stigProcessed: number;
+  newStigCount: number;
+  updatedStigCount: number;
+  unchangedStigCount: number;
+  xmlsExtracted: number;
+};
+
+export const migrateBoundary = async (
+  boundaryId: number,
+  newStigLibraryId: number,
+): Promise<{ results: string }> => {
+  try {
+    const boundarySystems = await System.findAll({ where: { BoundaryId: boundaryId } });
+    for (const system of boundarySystems) {
+      const currentAssessments = await Assessment.findAll({
+        where: {
+          SystemId: system.id,
+          succeededByAssessmentId: { [Op.is]: null },
+        },
+      });
+      for (const assessment of currentAssessments) {
+        const oldStig = await Stig.findOne({ where: { id: assessment.StigId } });
+        const newStig = await Stig.findOne({
+          where: { stigid: oldStig.stigid },
+          include: [
+            {
+              model: StigLibrary,
+              where: { id: newStigLibraryId },
+              required: true,
+            },
+          ],
+        });
+
+        await system.removeStig(oldStig);
+        await system.addStig(newStig);
+
+        const newAssessment = await createBlankAssessment(assessment.SystemId, newStig.id);
+
+        assessment.succeededByAssessmentId = newAssessment.id;
+        await assessment.save();
+
+        const oldChecks = await AssessmentItem.findAll({
+          where: { AssessmentId: assessment.id },
+          include: [{ model: StigData, attributes: ["vuln_num", "rule_id"] }],
+        });
+        const newChecks = await AssessmentItem.findAll({
+          where: { AssessmentId: newAssessment.id },
+          include: [{ model: StigData, attributes: ["vuln_num", "rule_id"] }],
+        });
+
+        for (const check of newChecks) {
+          const matchingCheck = findMatch(check, oldChecks);
+
+          if (matchingCheck) {
+            check.status = matchingCheck?.status;
+            check.comments = matchingCheck?.comments;
+            check.finding_details = matchingCheck?.finding_details;
+            check.severity_override = matchingCheck?.severity_override;
+            check.severity_justification = matchingCheck?.severity_justification;
+            check.AssessmentId = newAssessment.id;
+            await check.save();
+          }
+        }
+      }
+    }
+    return { results: "Success" };
+  } catch (error) {
+    logger.error(error);
+    throw error;
+  }
+};
+
+export const findMatch = (check: AssessmentItem, array: AssessmentItem[]) => {
```

Comment:
nitpick: i like having functions that are depended upon be above their users, so in this case 'findmatch' is depended upon by 'migrateboundary' so it would be above 'migrateboundary'.